### PR TITLE
Add TwoInputBenchmark

### PR DESCRIPTION
**Pull request by sunhaibotb**
Adds a high level  benchmark based on DataStream API for TwoInputStreamOperator.

### CODE REVIEW

File: src/main/java/org/apache/flink/benchmark/TwoInputBenchmark.java [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+	}
+
+	@Benchmark
+	public void mapSink(FlinkEnvironmentContext context) throws Exception {
+
+		StreamExecutionEnvironment env = context.env;
+		env.enableCheckpointing(CHECKPOINT_INTERVAL_MS);
+		env.setParallelism(1);
+
+		DataStreamSource<Long> source1 = env.addSource(new LongSource(RECORDS_PER_INVOCATION));
```
Comment (pnowojski) [inline src/main/java/org/apache/flink/benchmark/TwoInputBenchmark.java]: `RECORDS_PER_INVOCATION / 2` to have the total number or records inlined? However I'm not sure what should be our convention here.

File: src/main/java/org/apache/flink/benchmark/TwoInputBenchmark.java [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+		Options options = new OptionsBuilder()
+			.verbosity(VerboseMode.NORMAL)
+			.include(".*" + TwoInputBenchmark.class.getSimpleName() + ".*")
+			.build();
+
+		new Runner(options).run();
+	}
+
+	@Benchmark
+	public void mapSink(FlinkEnvironmentContext context) throws Exception {
```
Comment (pnowojski) [inline src/main/java/org/apache/flink/benchmark/TwoInputBenchmark.java]: This name is colliding with already existing benchmark. I've added an explanation [to the README](https://github.com/dataArtisans/flink-benchmarks/#naming-convention) about the naming convention/restrictions that we have, please take a look at it.

Maybe `twoInputMapSink`?

**Review by pnowojski — CHANGES_REQUESTED** [review]
Thanks @sunhaibotb. I've left two smaller comments.

After resolving the comments, could you provide the output of running the new benchmark with results and how much did the benchmarking take time?

File: src/main/java/org/apache/flink/benchmark/TwoInputBenchmark.java (hunk 1 above)
Comment (sunhaibotb) [reply src/main/java/org/apache/flink/benchmark/TwoInputBenchmark.java]: > RECORDS_PER_INVOCATION / 2 to have the total number or records inlined?

You are right.

**Comment by sunhaibotb** [comment]
I have update the code.  The benchmarking takes `02:22 minutes` when `RECORDS_PER_INVOCATION` is `15,000,000`,  and the output of benchmarking is as follow. @pnowojski 

```
Result "org.apache.flink.benchmark.TwoInputBenchmark.twoInputMapSink":
  8014.500 ±(99.9%) 465.530 ops/ms [Average]
  (min, avg, max) = (6150.633, 8014.500, 9516.431), stdev = 696.783
  CI (99.9%): [7548.970, 8480.029] (assumes normal distribution)


# Run complete. Total time: 00:02:08

Benchmark                           Mode  Cnt     Score     Error   Units
TwoInputBenchmark.twoInputMapSink  thrpt   30  8014.500 ± 465.530  ops/ms
```

**Comment by sunhaibotb** [comment]
I changed `RECORDS_PER_INVOCATION` from `15,000,000` to `25,000,000`, and found that the throughput increased significantly (which means `15,000,000` is not enough). Then increased it to `30,000,000`, the throughput is essentially no longer increasing. Therefore, I finally decided to change `RECORDS_PER INVOCATION` to `25,000,000`, and the code has been update. The benchmarking takes about `3 minutes` when `RECORDS_PER_INVOCATION` is `25,000,000`.

**Review by pnowojski — APPROVED** [review]
LGTM

**Pull request merged**